### DISCUSSION TITLE
Support to generate providers as static method of class.

### DISCUSSION THREAD

**Discussion by Kurogoma4D** [rrousselGit/riverpod#3626] [Ideas]
Since I saw #814 before, I've been making providers using this method sometimes.
I'm wondering if there is a way for code generation to support it in the future.

At least, I know that a function-based providers would be able to implement like this:
```dart
class SomeFeatureProviders {
  @riverpod
  static int foo(FooRef ref) => 42;
}
```

And there are issues with how the Notifiers is written.

**Comment by rrousselGit** [comment]
It's just not doable without macros or static extensions.
